gui/MainWindow.ui

```diff
@@ -646,9 +646,9 @@
    <container>1</container>
   </customwidget>
   <customwidget>
-   <class>MetadataView</class>
+   <class>kwiver::arrows::qt::MetadataView</class>
    <extends>QScrollArea</extends>
-   <header>MetadataView.h</header>
+   <header location="global">arrows/qt/widgets/MetadataView.h</header>
```

Comment:
@mwoehlke-kitware It occurs to me that qtExtension is providing widgets to TeleSculptor but these do no need the `location="global"` or the full path to make them work.  What does qtExtensions do to provide widgets that appear to be local to TeleSculptor?

Reply:
If qtExtensions headers aren't using `global`, that's an error/oversight; they ought to be.

That said, this just gets literally translated into a `#include`; `global` just means `#include <path>` instead of `#include "path"`. Accordingly, in most projects, headers from external libraries (including qtExtensions) should be `global`. OTOH, things will probably work either way; I believe `#include <path>` vs. `#include "path"` only makes an *actual* difference in some instances where the order of include path search may produce a different absolute path. (I'm pretty sure I've even seen projects use `""`-quoted paths for external headers, although that's not a style I would recommend.) So, strictly speaking, you probably don't *need* the `global` here, but it is recommended.

As for the path, that's just different conventions for what include paths are provided by the imported library. qtExtensions provides a bunch of paths with subpaths such that you can write e.g. `#include <qtThrobber.h>`, whereas KWIVER only provides the top-level include directory and expects you to write e.g. `#include <vital/types.h>`.

One might argue that qtExtensions shouldn't do this, but its headers also follow a naming convention that is less likely to have external clashes, unlike e.g. `#include <types.h>`. Since I would not recommend changing KWIVER's overall policy to *not* require (project-root-relative) paths, I also can't recommend making exceptions to it just for the Qt arrow. This is just the way life is; some libraries use partial paths (KWIVER), and some don't (Qt, qtExtensions).

Reply:
Thanks.  That makes sense.  I'm fine with leaving it as is.  I was just worried that I was doing it "wrong".

Reply:
Nope, looks exactly like I'd expect :slightly_smiling_face:. (If there are qtExtensions ones that *aren't* `global`, feel free to fix them.)